stor/dx.py

```diff
@@ -144,41 +138,273 @@ def download_objects(self):  # may not need it
         raise NotImplementedError
 
     def remove(self):
-        raise NotImplementedError
+        """Removes a single object from DX platform
+
+        Raises:
+            ValueError: The path is invalid.
+        """
+        if not self.resource:
+            raise ValueError('DXPath must point to single data object to call remove')
+        file_handler = dxpy.DXFile(dxid=self.canonical_resource,
+                                   project=self.canonical_project)
+        file_handler.remove()
 
     def rmtree(self):
-        raise NotImplementedError
+        """
+        Removes a resource and all of its contents. The path should point to a project or directory.
+
+        Raises:
+            ValueError: The path points to a file
+        """
+        proj_handler = dxpy.DXProject(self.canonical_project)
+        if not self.resource:
+            return proj_handler.destroy()
+        try:
+            proj_handler.remove_folder(self.resource, recurse=True)
+        except dxpy.exceptions.ResourceNotFound as e:
+            raise NotFoundError('No folders were found with the given path ({})'
+                                .format(self), e)
+
+    def makedirs_p(self):
+        """Make directories from DX folder paths.
+
+        The resulting folders will have same access permissions as the project.
+        """
+        if not self.resource:
+            if not self.exists():
+                raise ValueError('Project ({}) cannot be created with makedirs_p'.format(self))
+            return
+        proj_handler = dxpy.DXProject(self.canonical_project)
+        proj_handler.new_folder(self.resource, parents=True)
 
     def isdir(self):
-        if not self.resource or self._is_folder():
-            return self.exists()
-        return False
+        """Determine if path is an existing directory
+
+        Returns (bool) : True if path is an existing folder path or project
+        """
+        if not self.resource and self.exists():  # path could be a project
+            return True
+        # or path could be a folder
+        try:
+            self.listdir()
+            return True
+        except NotFoundError:
+            return False
 
     def isfile(self):
+        """Determine if path is an existing file
+
+        Returns (bool) : True if path points to an existing file
+        """
+        if not self.resource or self.endswith('/'):
```

Comment:
So, the only promise 'isfile' makes is that it will return True if the path exists as a file on DX platform, and False everywhere else. Paths like `dx://proj:/file/` automatically refer to a folder, because of the trailing slash and hence can readily return False.